client/network/server-url.ts

```diff
@@ -12,6 +9,14 @@ export function makeServerUrl(path: string) {
     return path
   }
 
+  // This can't be a top-level variable since we're importing a module that's using this function
```

Comment:
I very much do not understand this comment, nor the semantics that you think are at play here. If we're importing a file that is using this file, we have fucked up, and we should fix that instead of doing this. Circular dependenices are bad